documentation/specs/FormLayout.md

```diff
@@ -0,0 +1,334 @@
+# `FormLayout` Component Specification
+
+## Overview
+
+`FormLayout` arranges a layout of form elements with preset spacing.
+
+### Prior Art
+
+- [Paste `<Form />`](https://paste.twilio.design/components/form)
+
+---
+
+## Design
+
+`FormLayout` will be a compound component consisting of `FormLayout`, `FormLayout.Header`, `FormLayout.Title`, `FormLayout.HelperText`, `FormLayout.Section`, and `FormLayout.Grid`.
+
+`FormLayout` is a polymorphic component. Using the `as` prop, `FormLayout` can render as a `form` HTML element. By default, it will render as a `div` to not presuppose consumer structure.
+
+`FormLayout.Section` can be used to layout forms in sections. Sections are rendered as `fieldset`s and their titles rendered as `legend`s.
+
+`FormLayout.Header`, `FormLayout.Title`, and `FormLayout.HelperText` exist to provide headers, titles, and helper text to forms. These can be used within the main form layout as well as in sections.
+
+`FormLayout.Grid` can be used to layout the form in columns. This component is a simple wrapper around `HorizontalGrid` with preconfigured gap settings.
+
+### API
+
+```ts
+export type FormLayoutProps<T extends ElementType = "div"> =
+  ComponentProps<T> & {
+    as?: T;
+    children: ReactNode;
+  };
+
+type FormLayoutHeaderProps = {
+  children: ReactNode;
+};
+
+type FormLayoutTitleProps = {
+  id?: string;
+  as?: "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+  children: ReactNode;
+};
+
+type FormLayoutHelperTextProps = {
+  children: ReactNode;
+};
+
+type FormLayoutSectionProps = {
+  children: ReactNode;
+};
+
+type FormLayoutGridProps = {
+  children: ReactNode;
+  columns: HorizontalGridProps["columns"];
+};
```

Comment:
**question** is the intention for the grid to be more or less a lightweight wrapper around `HorizontalGrid`?

Reply:
yep exactly. it's a lightweight wrapper around `HorizontalGrid` with gap settings configured to the form layout design requirements